arch/common/sw_isr_common.c

```diff
@@ -16,12 +16,30 @@
 
 #ifdef CONFIG_MULTI_LEVEL_INTERRUPTS
 
-struct irq_parent_offset {
-	unsigned int irq;
-	unsigned int offset;
-};
+/* Filter out interrupt controllers with `has-multi-instance` prop */
+#define _FILTER_INTC(node_id)                                                                      \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, interrupt_controller),                               \
+		    (DT_PROP_OR(node_id, has_multi_instance, 0)), (0))
```

Comment:
TL;DR - The one issue I have here is with `has_multi_instance`. DT is implicitly multi-instance. Instead of using `has_multi_instance`, I would suggest rather using something like `DT_NUM_INST_STATUS_OKAY()` which conveys multi-instance status without an additional DT property. The number of instances of a dt-compat is implicitly part of DT, so it's not really necessary to explicitly add this property that should always be true.